dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,15 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory() const
+    {
+        IPropertyTree *history = root->queryPropTree("History");
+        //It can be null if the file created with an older version
+        if (!history)
+            history = root->setPropTree("History",createPTree("History")); // takes ownership
```

Comment:
From  Bjarne Stroustrup The C++ Programming Language, 4th Edition (Clasese, page 464):
"Note that const does not apply (transitively) to objects accessed through pointers or references.
The human reader may consider such an object as ‘‘a kind of subobject,’’ but the compiler does not know such pointers or references to be any different from any others. That is, a member pointer does not have any special semantics that distinguish it from other pointers."